src/uu/tr/src/operation.rs

```diff
@@ -319,13 +329,31 @@ impl Sequence {
             recognize(many_m_n(1, 3, one_of("01234567"))),
             |out: &[u8]| {
                 let str_to_parse = std::str::from_utf8(out).unwrap();
+                match u8::from_str_radix(str_to_parse, 8) {
+                    Ok(ue) => Some(ue),
+                    Err(_pa) => None,
+                }
+            },
+        )(input)
+    }
 
+    fn parse_octal_up_to_three_digits_with_warning(input: &[u8]) -> IResult<&[u8], u8> {
+        map_opt(
+            recognize(many_m_n(1, 3, one_of("01234567"))),
+            |out: &[u8]| {
+                let str_to_parse = std::str::from_utf8(out).unwrap();
                 match u8::from_str_radix(str_to_parse, 8) {
                     Ok(ue) => Some(ue),
                     Err(_pa) => {
-                        // TODO
-                        // A warning needs to be printed here
-                        // See https://github.com/uutils/coreutils/issues/6821
+                        let origin_octal: &str = std::str::from_utf8(input).unwrap();
+                        let actual_octal_tail: &str = std::str::from_utf8(&input[0..2]).unwrap();
+                        let outstand_char: char = char::from_u32(input[2] as u32).unwrap();
+                        show_warning!(
+                            "the ambiguous octal escape \\{} is being\n        interpreted as the 2-byte sequence \\0{}, {}",
+                            origin_octal,
+                            actual_octal_tail,
+                            outstand_char
+                        );
                         None
                     }
                 }
```

Comment:
Here it might be an option to use something like:
```rust
let result = u8::from_str_radix(str_to_parse, 8).ok();

if result.is_none() {
     let origin_octal: &str = std::str::from_utf8(input).unwrap();
     let actual_octal_tail: &str = std::str::from_utf8(&input[0..2]).unwrap();
     let outstand_char: char = char::from_u32(input[2] as u32).unwrap();
     show_warning!(
         "the ambiguous octal escape \\{} is being\n        interpreted as the 2-byte sequence \\0{}, {}",
         origin_octal,
         actual_octal_tail,
         outstand_char
     );
}

result
```
This way you could remove one indentation level.

Reply:
I don't think this is necessary. But it does make the code looks nicer. I think I'll accept this suggestion.